docs/rfcs/1015-auto-enrollment-device.md

```diff
@@ -0,0 +1,135 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+# Auto-Enrollment Device
+
+## 1 - Goals
+
+1. Allow simplified use of multiple devices by not requiring device-to-device enrollment.
+2. Provide configuration for per-organization opt-out of this solution for security reasons.
+
+## 2 - Overview
+
+## 2.1 - Different types of devices
+
+In Parsec, a device is supposed to:
+
+- Correspond to a single physical machine (e.g. desktop, phone, etc.)
+- Never run in parallel (as it messes with the sync system)
+
+However, with Parsec account, we introduce one exception to this: the "Auto-Enrollment Device".
+The idea is to have a special device that only lives on the server and is used to
+create new devices (hence bypassing the need for device-to-device enrollment).
+
+This device's `LocalDevice` is stored server-side in the Account Vault (see RFC 1014).
+
+Since the Account Vault can be disabled on a per-organization basis (for security
+reason, see RFC 1014), this feature is work in a best effort basis (i.e. traditional
+device-to-device enrollment is still used otherwise).
+
+### 2.2 - Relationship with existing enrollment
+
+During the enrollment process, the claimer's client should decide what to do with
+the new device.
+
+This is done solely based on the organization configuration:
+
+- If the organization allows Account Vault, the new device is an Auto-Enrollment Device.
+  It is uploaded to the server and then used to create a new device that will be the
+  actual device used locally.
+- Otherwise, the new device is the actual device used locally.
```

Comment:
What if the configuration is changed, e.g. Account Vault is first enabled then disabled (or viceversa). This shouldn't be so frequent, but we should consider how the system would behave and manage compatibility with existing (local or remote) devices.

Reply:
IMO it should behave like a fuse, meaning that it can only be enabled and never disabled.

Disabling the Account feature pose the question what to do with the device on account server and for the devices created from them for a given organization.

Reply:
RFC has been renamed "Registration device"